Tests/MoyaProviderSpec.swift

```diff
@@ -811,6 +811,70 @@ class MoyaProviderSpec: QuickSpec {
                 expect(completedValues.last) == true // the last must be true
                 expect(progressObjects.filter { $0 != nil }.count) == progressObjects.count // no nil object
             }
+
+            it("tracks progress of multipart request") {
+
+                let url = Bundle(for: MoyaProviderSpec.self).url(forResource: "testImage", withExtension: "png")!
+                let string = "some data"
+                guard let data = string.data(using: .utf8) else { fatalError("Failed creating Data from String \(string)") }
+                let target: HTTPBin = .uploadMultipart([
+                    MultipartFormData(provider: .file(url), name: "file", fileName: "testImage"),
+                    MultipartFormData(provider: .data(data), name: "data")
+                    ], nil)
+
+                var progressObjects: [Progress?] = []
+                var progressValues: [Double] = []
+                var completedValues: [Bool] = []
+                var error: MoyaError?
+
+                waitUntil(timeout: 5.0) { done in
+                    let progressClosure: ProgressBlock = { progress in
+                        progressObjects.append(progress.progressObject)
+                        progressValues.append(progress.progress)
+                        completedValues.append(progress.completed)
+                    }
+
+                    let progressCompletionClosure: Completion = { (result) in
+                        switch result {
+                        case .failure(let err):
+                            error = err
+                        case .success(let response):
+                            do {
```

Comment:
Huh, this one I'm having trouble with (kinda a bit too complicated for me). Do you think you could simplify this one a little bit?

Reply:
I can remove all code about "progress" but the test I think must be elsewhere
Just a test (or many tests in one séparantes file) about doing an upload with multipart data. 

I see anything else to simplify.

Reply:
I can remove all code about "progress" but the test I think must be elsewhere
Just a test (or many tests in one séparantes file) about doing an upload with multipart data. 

I see anything else to simplify.

Reply:
Oh, I think I understand now. We are checking two things (correct me if I'm wrong):
1. Check if the progress is tracked correctly.
2. Check if the response is correct as well.

I think that we don't really have to test the response here (2.), as this is just probably sample data test and if it's returned correctly in the stubbed response.  If there is no such test yet (not sure now, would have to check), we could introduce it, but separately. 

What do you think?

Reply:
Httpbin.org return a beautiful JSON output with all the form data that I post.

So I check that all multipart data I post, are in the JSON response.
I check that the content uploaded is exact ie. same string, have a file

Yes, separated tests could be done.

Reply:
Hmm, we should still split it, because in this test we check only for progress of the upload. In the next one we could check if the upload was correct (checking the JSON). Also we should somehow simplify this if/let hell.